actions/check_branch_and_release_type/action.yaml

```diff
@@ -0,0 +1,23 @@
+name: Checks the branch and release type
+description: 'Enforces that pre-release builds are done on a release branch and release builds are done on the default branc'
+inputs:
+  release_type:
+    description: The type of version number to return. Must be one of [Snapshot, Patch, Minor or Major]
+    required: true
+runs:
+  using: "composite" 
+
+  steps:
+    - name: Fail if Pre-release on Default branch
+      if: ${{ inputs.release_type == 'Snapshot' && github.event.repository.default_branch == github.ref_name }}
+      uses: actions/github-script@v7
+      with:
+          script: |
+            core.setFailed('Snapshot packages can not be created on the default branch. Release Type: ${{ inputs.release_type }}, Branch: ${{ github.ref_name }}')
+
+    - name: Fail if Release and not on Default branch
+      if: ${{ inputs.release_type != 'Snapshot' && github.event.repository.default_branch != github.ref_name }}
```

Comment:
Should this consider the `force_release` flag?

Reply:
force_release is really only for testing of builds, not code, so I don't think you should be able to bypass this.